### PR TITLE
[css-images-4][css-backgrounds-4] Added stripes() function

### PR DESCRIPTION
**Pull request by SebastianZ**
This change is meant to convert the [resolution of #2532](https://github.com/w3c/csswg-drafts/issues/2532#issuecomment-402327492) into spec. text.

It adds the `stripes()` function as a generally usable function as discussed in #2532 to CSS Images 4.

Note that I changed the syntax slightly to incorporate the `fr` unit mentioned in @LeaVerou's description. I did that by adding the `<flex>` value defined in CSS Grid 1, though the definition there currently focuses on the usage within grid layout. So its description either might need to be changed/generalized or the unit needs to be defined again within CSS Images 4. I'd prefer to generalize it (and maybe even move it to CSS Values 4).

The `stripes()` function is then used as a replacement for the colors list defined for `border-*-color` in CSS Backgrounds 4.

Reading through the discussions, a few things were still unclear to me or were discussed but not resolved on.

1. What's the behavior when the width of the image the `stripes()` function is used in is smaller or bigger? I've defined that the rest is transparent when the stripes thickness is smaller and that the stripes are clipped when they are bigger than the image, as that seems to be what people were in favor of in the discussion.
2. Should the stripes be painted inwards or outwards? For now I kept the definition of "starting outwards", though to me it seems a bit counter-intuitive that the borders go inwards. Also, the wording may be clarified further.
3. Should there be a description of how stripes (or generally `border-color`) interact with `border-style`? Clipping was mentioned in the discussion, also treating `inset`, `outset`, `groove`, and `ridge` as solid.

Also note that this change is based on #7014, because without those changes Bikeshed won't compile the spec.

Sebastian

### CODE REVIEW

**Comment by svgeesus** [comment]
@LeaVerou @fantasai we resolved to add `stripes()` ages ago and the edit was not made, this PR looks like a good way forward?

**Comment by LeaVerou** [comment]
Hi Sebastian,

Thank you for working on this.
It has been a long time, but IIRC the resolution of these discussions, we would define 1 dimensional images, and `stripes()` would just produce a 1 dimensional image. Right now in CSS we have 2d images (gradients, urls etc), and 0 dimensional images (`image(<color>)`). I see that in this PR, `stripes()` is defined as its own type, and can only be used in properties that explicitly allow it. We'd like to be a bit more general than that.

**Comment by SebastianZ** [comment]
Thank you for reviewing the PR! There are obviously some open questions, so let's [continue the discussion in the issue](https://github.com/w3c/csswg-drafts/issues/2532#issuecomment-1040238169) before we continue with this PR.

Sebastian

**Comment by SebastianZ** [comment]
@LeaVerou @tabatkins I went ahead and interpreted your statements in the issue to make the relevant changes. That means, I added `<stripes()>` to `<image>` and changed its description.

I also added it to `outline-color` as previously resolved on.  I've kept the reference to `<flex>` for now. Though as pointed out earlier, I think its definition should be generalized and moved to a more general place to make it clear that it doesn't only apply to grid layout anymore.

I am awaiting your feedback on these changes.

Sebastian

**Comment by SebastianZ** [comment]
As noted by @LeaVerou, I've now put `stripes()` behind a `<1d-image>` type, so it can easily be reused and extended.

Open questions:

* Is it ok to allow different orders in the syntaxes?
* What to do with `<flex>`?

Sebastian

**Comment by SebastianZ** [comment]
>     * Is it ok to allow different orders in the syntaxes?

I used the [double pipe](https://drafts.csswg.org/css-values-4/#comb-any) in the syntaxes, i.e. `<color> || <1d-image>`. This allows the two values to be in any order.

That allows to specify something like `border-color: red stripes(white, silver) stripes(white, silver) yellow lime stripes(black, gray) blue;`.

We might use juxtaposing instead, i.e. `<color> <1d-image>` to force the order of first `<color>` then `<1d-image>`.

>     * What to do with `<flex>`?

As noted in a [previous comment](https://github.com/w3c/csswg-drafts/pull/7029#issuecomment-1068988402), I reused `<flex>` from CSS Grid 1 because I agree with you that it fits very well this use case.

Though, as it is now used in two completely different places, should it's description be generalized and be moved somewhere else? And if so, should that be done in this PR already?

Sebastian

**Comment by LeaVerou** [comment]
> > ```
> > * Is it ok to allow different orders in the syntaxes?
> > ```
> 
> I used the [double pipe](https://drafts.csswg.org/css-values-4/#comb-any) in the syntaxes, i.e. `<color> || <1d-image>`. This allows the two values to be in any order.
> 
> That allows to specify something like `border-color: red stripes(white, silver) stripes(white, silver) yellow lime stripes(black, gray) blue;`.
> 
> We might use juxtaposing instead, i.e. `<color> <1d-image>` to force the order of first `<color>` then `<1d-image>`.

Oh no, you need to use a single pipe: `[ <<color>> | <<1d-image>> ]`. It's a `<color>` OR `<1d-image>` but not both. It doesn't make sense to have `border-color: red stripes(white, silver) stripes(white, silver) yellow lime stripes(black, gray) blue;` because you can't tell what is for what side.

> > ```
> > * What to do with `<flex>`?
> > ```
> 
> As noted in a [previous comment](https://github.com/w3c/csswg-drafts/pull/7029#issuecomment-1068988402), I reused `<flex>` from CSS Grid 1 because I agree with you that it fits very well this use case.
> 
> Though, as it is now used in two completely different places, should it's description be generalized and be moved somewhere else? And if so, should that be done in this PR already?

Yeah, it should go to css-values. @fantasai @tabatkins ?

**Comment by tabatkins** [comment]
Yeah we can move `<flex>` out to the general V&U spec now that it's being used in more than just Flexbox.

**Comment by LeaVerou** [comment]
> Yeah we can move `<flex>` out to the general V&U spec now that it's being used in more than just Flexbox. (I'll wait for merging of this PR tho.)

Great, should we call it something else in that case? `<fraction>`?

File: css-backgrounds-4/Overview.bs
Comment (fantasai) [inline css-backgrounds-4/Overview.bs:213]: What happens if no color is specified? Do we get transparent or currentColor or...?

File: css-backgrounds-4/Overview.bs
Comment (fantasai) [inline css-backgrounds-4/Overview.bs:235]: I believe this example is trying to show off the use of the 'border-color' shorthand, this rewrite into longhands isn't appropriate.

File: css-backgrounds-4/Overview.bs
Comment (fantasai) [inline css-backgrounds-4/Overview.bs:216]: Interpolation should probably be defined in the definition of the `stripes()` function, rather than here. And it should match what we do for gradients.

File: css-images-4/Overview.bs
Comment (fantasai) [inline css-images-4/Overview.bs:75]: I don't think we have a WG resolution on using 1D images as 2D images, so I wouldn't include that here.

(I also don't think it's a useful idea. If we want stripes() to work in 2D, we should define a 2D stripes function, which can re-orient the stripes as appropriate. It's not worth the implementation/testing/documentation effort to have a stripes() function available in all 2D contexts that can only do horizontal stripes. IMHO)

File: css-images-4/Overview.bs
Comment (fantasai) [inline css-images-4/Overview.bs:1911]:
```suggestion
	The ''stripes()'' function allows defining a one-dimensional image
	consisting of stripes of different colors with different widths.
```

File: css-images-4/Overview.bs
Comment (fantasai) [inline css-images-4/Overview.bs:1913]:
```suggestion
	The syntax for <<1d-image>> and <<stripes()>> is defined as:
```

File: css-images-4/Overview.bs
Comment (fantasai) [inline css-images-4/Overview.bs:1928]: I'd suggest using the DL format we use in property value definitions. Use dfn-for="stripes()".

File: css-images-4/Overview.bs
Comment (fantasai) [inline css-images-4/Overview.bs:1924]: There's no usefulness in fr values being equivalent to percentages. What you want is to treat them similar to grid, i.e. subtract out all the fixed-size stripes and distribute the remaining space accordingly. If the remaining space is not defined, treat it as zero.

File: css-images-4/Overview.bs
Comment (fantasai) [inline css-images-4/Overview.bs:1939]: No, we don't want to remove the fr values. :) We just want lengths and colors computed. So the computed value of this function is a list, each item pairing a computed color and either a `<<flex>>` value or a computed `<<length-percentage>>`.

File: css-images-4/Overview.bs
Comment (fantasai) [inline css-images-4/Overview.bs:1908]: Start this section by defining what a one-dimension image is. Then mention that the only one-dimensional image type defined in this level is `stripes()`, and go on to define it below.

File: css-ui-4/Overview.bs
Comment (fantasai) [inline css-ui-4/Overview.bs:297]: I suggest using the same sentence construction as the `<<color>>` type, for consistency.

File: css-backgrounds-4/Overview.bs
Comment (fantasai) [inline css-backgrounds-4/Overview.bs:207]: Suggested rewrite:

```suggestion
	The stripes defined by <<1d-image>> follow the shape of the border
	on the side to which they apply,
	and are drawn in bands starting from the [=padding edge=] and progressing outwards.
```

File: css-backgrounds-4/Overview.bs
Comment (fantasai) [inline css-backgrounds-4/Overview.bs:211]: I'd use `the specified <<color>>` rather than `a given <<color>>`.

**Review by fantasai — DISMISSED** [review]
Overall looks pretty good. Posted some specific things to fix.

Also, the line-breaking convention for our specs is https://rhodesmill.org/brandon/2012/one-sentence-per-line/ (This is why the lines look so ragged, there's a method to the madness.)

(You might also find https://lists.w3.org/Archives/Public/spec-prod/2018OctDec/0011.html a useful read as background context, though nothing else there is particularly relevant to this PR.)

**Comment by tabatkins** [comment]
I've done what turned out to be a decent bit of rearranging and rewording on the Images part. Largely editorial; the only normative changes are to handle sums of flexes < 1 the same way flexbox does, and define the painting area when used as a 2d image.

**Comment by SebastianZ** [comment]
> > > ```
> > > * Is it ok to allow different orders in the syntaxes?
> > > ```
> > 
> > 
> > I used the [double pipe](https://drafts.csswg.org/css-values-4/#comb-any) in the syntaxes, i.e. `<color> || <1d-image>`. This allows the two values to be in any order.
> > That allows to specify something like `border-color: red stripes(white, silver) stripes(white, silver) yellow lime stripes(black, gray) blue;`.
> > We might use juxtaposing instead, i.e. `<color> <1d-image>` to force the order of first `<color>` then `<1d-image>`.
> 
> Oh no, you need to use a single pipe: `[ <<color>> | <<1d-image>> ]`. It's a `<color>` OR `<1d-image>` but not both. It doesn't make sense to have `border-color: red stripes(white, silver) stripes(white, silver) yellow lime stripes(black, gray) blue;` because you can't tell what is for what side.

I deliberately chose to allow both, so `<color>` serves as a fallback, as you noticed in an inline comment. `border-color` is one place where slashes would make it easier to distinguish between the different values like we have it for `grid-area`. So we could also express the syntax as:

```
<<color>>{1,4} | <<color>> <<1d-image>>? [ / <<color>> <<1d-image>>? ]{0,3}
```

That means, authors have to add the slashes when using `<1d-image>`, while the existing `<color>`-only syntax without slashes is still supported as legacy syntax. Applied to my example above, this would then look like

```
border-color: red stripes(white, silver) / yellow stripes(white, silver) / lime stripes(black, gray) / blue;
```

Sebastian

**Comment by SebastianZ** [comment]
> > Yeah we can move `<flex>` out to the general V&U spec now that it's being used in more than just Flexbox. (I'll wait for merging of this PR tho.)
> 
> Great, should we call it something else in that case? `<fraction>`?

As even Tab obviously mixed it up being part of Flexbox and not Grid, I'd say yes. 😉 Though I don't have a strong opinion on that.

And I agree with him that this should be done in a separate PR.

Sebastian

**Comment by SebastianZ** [comment]
> Overall looks pretty good. Posted some specific things to fix.
> 
> Also, the line-breaking convention for our specs is https://rhodesmill.org/brandon/2012/one-sentence-per-line/ (This is why the lines look so ragged, there's a method to the madness.)
> 
> (You might also find https://lists.w3.org/Archives/Public/spec-prod/2018OctDec/0011.html a useful read as background context, though nothing else there is particularly relevant to this PR.)

Thanks for the links! I'll correct the line-breaking.

Sebastian

File: css-backgrounds-4/Overview.bs
Comment (SebastianZ) [reply css-backgrounds-4/Overview.bs:213]: Good point! I'd say `currentcolor` as it's the initial value. I'll add that for now, though we may still discuss whether that's the best choice.

Sebastian

File: css-backgrounds-4/Overview.bs
Comment (SebastianZ) [reply css-backgrounds-4/Overview.bs:235]: Ok, I'll change it back to `border-color`.

Sebastian

File: css-images-4/Overview.bs
Comment (SebastianZ) [reply css-images-4/Overview.bs:75]: As pointed out in https://github.com/w3c/csswg-drafts/issues/2532#issuecomment-1104836974, the plan is to add 1D images to gradients. So we might end up using that instead of adding it to `<image>` directly.
Let's discuss that in the next call.

Sebastian

File: css-images-4/Overview.bs
Comment (SebastianZ) [reply css-images-4/Overview.bs]: By saying that `1fr` is removed I wanted to accomodate the serialization rules which say to omit component values for a shorter serialization when they don't change the meaning. And I defined that an omitted thickness is equivalent to `1fr`.
Though I guess it's not necessary to explicitly mention that as that's already part of the serialization rules.

I'll adjust the wording according to your suggestion.

File: css-images-4/Overview.bs
Comment (SebastianZ) [reply css-images-4/Overview.bs:1908]: Agree. I actually already had something prepared for that before that comment.

Sebastian

**Comment by SebastianZ** [comment]
> I've done what turned out to be a decent bit of rearranging and rewording on the Images part. Largely editorial; the only normative changes are to handle sums of flexes < 1 the same way flexbox does, and define the painting area when used as a 2d image.

I missed that when answering the previous comments. @tabatkins Thank you for the changes! With those you obviously already covered a few things noted by @fantasai.

I'll incorporate the rest of the changes and we can discuss the points that are still open afterwards.

Sebastian

**Comment by LeaVerou** [comment]
> > > > ```
> > > > * Is it ok to allow different orders in the syntaxes?
> > > > ```
> > > 
> > > 
> > > I used the [double pipe](https://drafts.csswg.org/css-values-4/#comb-any) in the syntaxes, i.e. `<color> || <1d-image>`. This allows the two values to be in any order.
> > > That allows to specify something like `border-color: red stripes(white, silver) stripes(white, silver) yellow lime stripes(black, gray) blue;`.
> > > We might use juxtaposing instead, i.e. `<color> <1d-image>` to force the order of first `<color>` then `<1d-image>`.
> > 
> > 
> > Oh no, you need to use a single pipe: `[ <<color>> | <<1d-image>> ]`. It's a `<color>` OR `<1d-image>` but not both. It doesn't make sense to have `border-color: red stripes(white, silver) stripes(white, silver) yellow lime stripes(black, gray) blue;` because you can't tell what is for what side.
> 
> I deliberately chose to allow both, so `<color>` serves as a fallback, as you noticed in an inline comment. `border-color` is one place where slashes would make it easier to distinguish between the different values like we have it for `grid-area`. So we could also express the syntax as:
> 
> ```
> <<color>>{1,4} | [ <<color>>? <<1d-image>>? ]! [ / [ <<color>>? <<1d-image>>? ]! ]{0,3}
> ```
> 
> That means, authors have to add the slashes when using `<1d-image>`, while the existing `<color>`-only syntax without slashes is still supported as legacy syntax. Applied to my example above, this would then look like
> 
> ```
> border-color: red stripes(white, silver) / yellow stripes(white, silver) / lime stripes(black, gray) / blue;
> ```
> 
> Sebastian

What kind of fallback? If `stripes()` is not supported, the entire declaration would be dropped, including the color.
If we want `stripes()` to accept an optional color to paint the stripes over, it should probably be part of the syntax for `stripes()`, not in each individual property's grammar as separate tokens. @fantasai any thoughts on this?

File: css-images-4/Overview.bs
Comment (tabatkins) [reply css-images-4/Overview.bs]: Computed value is not serialization! The two concepts are completely unrelated, tho they were often conflated in older spec text. In this case, every stripe has a color and a thickness; the fact that a `1fr` thickness is omitted in the serialization falls out automatically from the shortest serialization principle.

**Comment by tabatkins** [comment]
I don't know what the "color to paint the stripes over" is for, tbh.

**Comment by LeaVerou** [comment]
> I don't know what the "color to paint the stripes over" is for, tbh.

No idea, but I assume that's what @SebastianZ intended for the fallback, a color to paint underneath any semi-transparent stripes. FWIW I don't think it's that necessary and complicates the syntax, so I'd be fine removing it.

**Comment by SebastianZ** [comment]
> > I don't know what the "color to paint the stripes over" is for, tbh.
> 
> No idea, but I assume that's what @SebastianZ intended for the fallback, a color to paint underneath any semi-transparent stripes. FWIW I don't think it's that necessary and complicates the syntax, so I'd be fine removing it.

Yes, the meaning of this "fallback color" was to let the color is drawn below any (semi-)transparent stripes. So, basically the same interaction as `background-color` with `background-image`. The idea behind that was that authors might expect that the `<color>` value still has an influence even when using `stripes()`.

Though I'm happy to change the syntaxes to `<color> | `<1d-image>` if we agree that this won't be the case and there are only use cases to have either `<color>` or `stripes()`.

Sebastian

**Comment by tabatkins** [comment]
Oh, I guess my comment I wrote yesterday on the `color || stripes` bit didn't actually end up getting posted - we literally can't do `||` anyway, because it makes `border-color` ambiguous if it has anywhere between 2 and 7 entries. `border-color: red stripes(green, blue)` might be one value (cloned to all four sides) or two values (red top and bottom, stripes left and right). Given that "just color" is allowed today, the only way to do it at all is with `[ <1d-image>? <color> ]+` - that is, a *required* color, with an optional 1d-image that *must* precede it.

But yeah, I don't understand the purpose of having both color and a 1d-image anyway. I understand the analogy you're drawing with background-color vs -image, but there's no use-case for this functionality in borders currently. More generally, we don't have the concept of layered borders like we have layered backgrounds.

**Comment by SebastianZ** [comment]
> Oh, I guess my comment I wrote yesterday on the `color || stripes` bit didn't actually end up getting posted - we literally can't do `||` anyway, because it makes `border-color` ambiguous if it has anywhere between 2 and 7 entries. `border-color: red stripes(green, blue)` might be one value (cloned to all four sides) or two values (red top and bottom, stripes left and right). Given that "just color" is allowed today, the only way to do it at all is with `[ <1d-image>? <color> ]+` - that is, a _required_ color, with an optional 1d-image that _must_ precede it.

You're right. I totally missed that ambiguity. Though I incidentally [brought up a solution for that earlier](https://github.com/w3c/csswg-drafts/pull/7029#issuecomment-1104969215) by requiring slashes between the values when `<1d-image>` is used.

> But yeah, I don't understand the purpose of having both color and a 1d-image anyway. I understand the analogy you're drawing with background-color vs -image, but there's no use-case for this functionality in borders currently. More generally, we don't have the concept of layered borders like we have layered backgrounds.

With the above said, I agree now to change it to `<color> | <1d-image>`. When use cases for layered borders arise, we can think about that again.

Sebastian

File: css-backgrounds-4/Overview.bs
Comment (SebastianZ) [reply css-backgrounds-4/Overview.bs]: Because we've now defined to have either a color or a 1D image that has transparent black as its default color, there's no need to define this anymore.

Sebastian

File: css-backgrounds-4/Overview.bs
Comment (SebastianZ) [reply css-backgrounds-4/Overview.bs]: Done.

File: css-backgrounds-4/Overview.bs
Comment (SebastianZ) [reply css-backgrounds-4/Overview.bs]: I've moved it to the "Interpolation" section in CSS Images 4 and tried to align its definition with the one for gradients. Please have a detailed look at it!

Sebastian

File: css-images-4/Overview.bs
Comment (SebastianZ) [reply css-images-4/Overview.bs]: As discussed, we've split the decision to use 1D images in 2D contexts out of this PR. For reference, see #7241 and #7244.

Sebastian

File: css-images-4/Overview.bs
Comment (SebastianZ) [reply css-images-4/Overview.bs]: This is covered by Tab's changes.

Sebastian

File: css-images-4/Overview.bs
Comment (SebastianZ) [reply css-images-4/Overview.bs]: I guess that's covered by Tab's changes already.

Sebastian

File: css-ui-4/Overview.bs
Comment (SebastianZ) [reply css-ui-4/Overview.bs]: I've done that. @fantasai Please have a look whether that's what you had in mind!

Sebastian

File: css-backgrounds-4/Overview.bs
Comment (SebastianZ) [reply css-backgrounds-4/Overview.bs]: Done.

File: css-backgrounds-4/Overview.bs
Comment (SebastianZ) [reply css-backgrounds-4/Overview.bs]: This is obsolete because we've changed the syntax to use either a color or a 1D image defining transparent black and its default color.

Sebastian

**Comment by SebastianZ** [comment]
I've removed `<1d-image>` from `<image>` as resolved and opened up separate issues for adding 1D images to 2D contexts. And I think I've addressed all remaining comments. So please give this PR another review and let me know if there's anything left!

Sebastian

File: css-images-4/Overview.bs
Comment (SebastianZ) [reply css-images-4/Overview.bs:1908]: That's already done by the changes made by @tabatkins.

Sebastian

**Comment by SebastianZ** [comment]
@fantasai @LeaVerou @tabatkins Is it possible that you give this another review? I'd really like to finish this.

Sebastian

**Comment by SebastianZ** [comment]
@bert-github I've added you as a reviewer, because you are an editor of CSS Backgrounds 4 and the others seem to be busy. It would be perfect if I got some feedback whether this patch can be merged now or if anything is still missing.

Sebastian

File: css-images-4/Overview.bs
Comment (SebastianZ) [reply css-images-4/Overview.bs]: I don't think this is still needed but please let me know if it is!

File: css-ui-4/Overview.bs
Comment (SebastianZ) [reply css-ui-4/Overview.bs]: I've fixed the two links now. The one to Color 4 was actually broken.

File: css-backgrounds-4/Overview.bs
Comment (SebastianZ) [reply css-backgrounds-4/Overview.bs:199]: For reference, here's the suggested syntax again:

```
<<color>>{1,4} | [ <<color>>? <<1d-image>>? ]! [ / [ <<color>>? <<1d-image>>? ]! ]{0,3}
```

And here's an example using that syntax:

```css
border-color: stripes(white, silver) / stripes(white, silver) / stripes(black, gray) / blue;
```

**Comment by SebastianZ** [comment]
As the comment list on this PR got quite long, let me summarize the pending points:

* I've moved the [definition for interpolating `<1d-image>` to Images 4](https://github.com/w3c/csswg-drafts/pull/7029#discussion_r864476143).
* I believe [fantasai's suggestion to add `dfn-for="stripes()"`](https://github.com/w3c/csswg-drafts/pull/7029#discussion_r854572936) is not needed anymore after all the changes.
* I've [previously made a suggestion](https://github.com/w3c/csswg-drafts/pull/7029#issuecomment-1104969215) to add slashes between the different values in the `border-color` shorthand to align it with other shorthands and make it future prove which I'd like to get feedback on.
* We agreed on moving `<flex>` to Values & Units (and possibly rename it to `<fraction>`) once this PR is merged.

@fantasai @tabatkins Could you please give this another review so the changes can be merged?

Sebastian

File: css-images-4/Overview.bs
Comment (Loirooriol) [inline css-images-4/Overview.bs:1976]: What does "If used generically as an `<image>`" mean, if  `<1d-image>` is no longer an `<image>`?

Comment (SebastianZ) [reply css-images-4/Overview.bs:1970]: That was just a left-over. I've removed the whole paragraph now as it didn't add any additional information anymore.

Thanks for the hint!

**Comment by SebastianZ** [comment]
I went ahead and merged the changes now as I believe all feedback was addressed and I waited for half a year for review.

I'll create separate issues for my suggestion to change the syntax of `border-color` and for generalizing the `<flex>` value. For reference, see [my previous comment](https://github.com/w3c/csswg-drafts/pull/7029#issuecomment-1272614499).

If there is still anything left, please let me know and I'll change it.

Sebastian

File: css-images-4/Overview.bs
Comment (Loirooriol) [inline css-images-4/Overview.bs:1933]: I think this is very confusing, the "painting area" is never defined but since it's an area it seems to be 2D. How to resolve a 1D `<length-percentage>` relatively to a 2D area?

Should this refer to the "length of the paint line" instead?

File: css-images-4/Overview.bs
Comment (Loirooriol) [reply css-images-4/Overview.bs:1970]: You removed another `used generically as an <<image>>`, but not this one.

File: css-images-4/Overview.bs
Comment (Loirooriol) [inline css-images-4/Overview.bs:2480]: If the two image***s***

File: css-images-4/Overview.bs
Comment (Loirooriol) [inline css-images-4/Overview.bs:2498]: How to interpolate a `<length-percentage>` with a `<flex>`?

Rather than checking that each image doesn't mix different types, should the requirement be that each pair of matched stripes have the same type?

File: css-images-4/Overview.bs
Comment (Loirooriol) [reply css-images-4/Overview.bs:1933]: https://github.com/w3c/csswg-drafts/issues/8160

File: css-images-4/Overview.bs
Comment (Loirooriol) [reply css-images-4/Overview.bs:1970]: https://github.com/w3c/csswg-drafts/issues/8161

File: css-images-4/Overview.bs
Comment (Loirooriol) [reply css-images-4/Overview.bs:2480]: https://github.com/w3c/csswg-drafts/issues/8162

File: css-images-4/Overview.bs
Comment (Loirooriol) [reply css-images-4/Overview.bs:2498]: https://github.com/w3c/csswg-drafts/issues/8163

**Pull request merged**
